install_config/configuring_admission_control.adoc

```diff
@@ -0,0 +1,161 @@
+link:../architecture/core_concepts/projects_and_users.html#namespaces[Namespace]
+must exist before creating resources in it. It also prevents creation of resources in a project
+that is in the process of getting deleted.
+
+* *BuildByStrategy* Restricts creation of
+link:../architecture/core_concepts/builds_and_image_streams.html#builds[Builds]
+and
+link:../dev_guide/builds.html#defining-a-buildconfig[BuildConfigs]
+to a strategy that the current user is
+link:../admin_guide/securing_builds.html[allowed to use].
+
+=== Overriding the list of admission control plugins for Kubernetes resources
+
+The list of admission control plugins for Kubernetes resources can be configured in the
+link:master_node_configuration.html#master-configuration-files[master-config.yaml]
+as follows:
+
+[source,yaml]
+----
+kubernetesMasterConfig:
+  admissionConfig:
+    pluginOrderOverride:
+    - NamespaceLifecycle
+    - OriginPodNodeEnvironment
+    - YourPlugin
+    - ...
+----
+
+The default list of admission control plugins for Kubernetes resources is:
+
+[source, yaml]
+----
+- NamespaceLifecycle
+- OriginPodNodeEnvironment
+- LimitRanger
+- ServiceAccount
+- SecurityContextConstraint
+- ResourceQuota
+- SCCExecRestrictions
+----
+
+* *NamespaceLifecycle* Enforces that a
+link:../architecture/core_concepts/projects_and_users.html#namespaces[Namespace]
+must exist before creating resources in it. It also prevents creation of resources in a project
+that is in the process of getting deleted.
+
+* *OriginPodNodeEnvironment* Ensures that a
+link:../architecture/core_concepts/pods_and_services.html#pods[Pod]'s node selector matches at least one node
+in the cluster.
+
+* *LimitRanger* Enforces resource usage
+link:../dev_guide/limits.html[limits] for pods created in a given Namespace.
+
+* *ServiceAccount* Ensures that all pods are associated with a valid
+link:../dev_guide/service_accounts.html[service account] and that the
+service account associated with the pod can properly execute it.
+
+* *SecurityContextConstraint* Ensures that
+link:../architecture/additional_concepts/authorization.html#security-context-constraints[security context constraints]
+that apply to the current user and pod service account are applied to a pod.
+
+* *ResourceQuota* Enforces
+link:../dev_guide/quota.html[resource quota] that has been configured for a particular project.
+
+* *SCCExecRestrictions* Prevents users from remotely executing processes on pods that they cannot create according to current
+link:../architecture/additional_concepts/authorization.html#security-context-constraints[security context constraints].
+
+== Providing configuration for Admission Control Plugins
+
+Certain admission control plugins may take an additional configuration file. The configuration for a plugin may be provided either
+as a file location or an embedded configuration object in
+link:master_node_configuration.html#master-configuration-files[master-config.yaml].
```

Comment:
On the startup of our master we write the embedded config to a temp file and pass that file's stream to the plugin.